controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -104,7 +104,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if err != nil {
 		logger.Error(err, "Failed to get Application from Component",
 			"Component.Name ", component.Name, "Component.Namespace ", component.Namespace)
-		return ctrl.Result{}, err
+		tknErr := tekton.AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(ctx, pipelineRun, r.Client, err)
+		if tknErr != nil {
+			return ctrl.Result{}, tknErr
+		}
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
 	}
 
 	if application == nil {
```

Comment:
Shouldn't we stop reconciliation also here? Because when component doesn't have application label, it will retry reconciliation as well.

(It's weird that a Get function can return nothing without returning error)

Reply:
isn't it actually just dead code?
if get returns NotFound error, how we can receive nil?

Reply:
Yes I think so. Let me add logic for stopping reconcilation.